fix for absolute file path in window

ccpu opened the pull request:
fixes #11

ccpu committed:
fix for absolute file path in window

ccpu committed:
fix for template import path

NewFuture commented:
@hardikmodha pls

ccpu committed:
storybook files

hardikmodha commented:
Thank you for your PR @Cyrus-d. I've been busy with my day job a lot lately, didn't get much time to take a look.

hardikmodha commented on src/create-tests/templates/TemplateManager.ts:
```diff
@@ -45,8 +45,8 @@ export class TemplateManager {
-
-        const importPath = path.join(path.relative(testFileDir, sourceFileDir), moduleName);
+        const relativePath = path.relative(testFileDir, sourceFileDir);
+        const importPath = [relativePath, moduleName].join('/');
```
I don't think we need this change.

ccpu committed:
testrunner

ccpu committed:
TestFileHelper

ccpu committed:
fixes

ccpu committed:
array of configs

ccpu committed:
rename, simplify

ccpu committed:
name

ccpu committed:
template
readme

ccpu committed:
github link

ccpu committed:
variables renaming

ccpu committed:
user inputs,

ccpu committed:
rename

ccpu committed:
terminal preserve focus

ccpu committed:
terminal cwd

ccpu committed:
shouldSwitchTerminalToCwd

ccpu committed:
read me

ccpu committed:
task label

ccpu committed:
fix task args

ccpu committed:
showMessageIfPathNotExist

ccpu committed:
userInputPrompt with none options

ccpu committed:
Merge branch 'master' of https://github.com/Cyrus-d/vscode-create-tests

ccpu committed:
fixed directory

ccpu committed:
dirty change

ccpu committed:
Readme change

ccpu committed:
activationEvents

ccpu committed:
added targetRootDirname and targetRootPath

ccpu committed:
fixed targetRootPath

ccpu committed:
added targetDirNameWithoutDirSuffix

ccpu committed:
fixed path

ccpu committed:
tempalte

Commit:
Improve error handling in TaskRunner and enhance replaceSourceDir utility function

ccpu committed:
feat(variable-resolver): add support for ${relativeFileToPackageRoot} variable

- Implemented logic to resolve the file path relative to the closest package root.
- Added fallback to absolute path if no package.json is found.
- Improved the `replaceSourceDir` function for better readability and maintainability.

ccpu committed:
Refactor code structure for improved readability and maintainability

ccpu committed:
Refactor code structure for improved readability and maintainability

ccpu committed:
Add unit tests for file generator utilities and setup Vitest configuration

- Implement tests for findRootDir, getFileName, isNewDirectory, isSupportExtension, switchToFile, and variable-resolver utilities.
- Create a test setup file to mock console methods and clear mocks before each test.
- Update tsconfig.json to include test files and adjust module resolution.
- Add Vitest configuration for testing environment, coverage reporting, and module aliasing.

ccpu committed:
fix(tsconfig): update rootDir and add tsconfig.build.json

* Change `rootDir` from `src` to `.` in `tsconfig.json` for better project structure.
* Introduce `tsconfig.build.json` to manage build-specific TypeScript configurations.

ccpu committed:
feat(variable-resolver): enhance variable resolution and add new utility functions

* Introduced `findFileDirUpwards` to search for files in parent directories.
* Added `getWorkspaceRoot` to retrieve the workspace root path.
* Updated variable resolution logic to support new variables:
  - `${outputFilePath}`
  - `${relativeFilePath}`
  - `${fileDirPath}`
  - `${fileExtension}`
  - `${fileNameWithExt}`
  - `${fileNameWithoutExt}`
  - `${relativeFileDir}`
  - `${relativeRootDirPath}`
  - `${workspaceToFileDir}`
  - `${rootDirNam}`
  - `${rootDirPath}`
  - `${filePathFromRoot}`
  - `${fileDirPathFromRoot}`
* Updated tests to cover new variable resolutions and utility functions.
* Refactored `launch.json` and `tasks.json` for improved debugging and build tasks.

ccpu committed:
refactor: update TypeScript configuration and improve file handling

- Changed TypeScript watch command to use the project root.
- Updated test command to include coverage reporting.
- Refactored `NewFileHelper` to simplify file location checks.
- Removed unused `find-root-dir` utility and replaced its functionality with `getWorkspaceRootForFile`.
- Enhanced `TaskRunner` to await file switching operations.
- Improved error handling in `switchToFile` utility.
- Removed deprecated tests related to `findClosestPackageRoot`.
- Added new tests for `getWorkspaceRoot` and `getWorkspaceRootForFile`.
- Updated Vitest configuration to improve coverage reporting.
